### PR TITLE
fix: add checkbox id name if not available based on label

### LINKED ISSUE #1448: Bug: user settings checkbox IDs are not set
### Describe the bug

Since the checkboxes has no IDs set right now, when you use multiple check boxes in a page only one of them is being triggered. 

https://github.com/open-sauced/insights/assets/18273833/dcf6a1d4-7090-482e-8526-bf819f411bfe

### Steps to reproduce

1. https://insights.opensauced.pizza/user/settings
2. click on: Display email on profile
3. you will see that Display current time is being checked. 

### Browsers

_No response_

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by a0m0rajab**
## Description

This PR fixes the similar IDs for the checkboxes in the whole repo, the previous behavior was using the same name for each checkbox, this PR will allow to generate a unique id based on the label.

**Important note:**: The GitHub UI is showing unnecessary changes due to changing spaces in the file, hiding spaces would show a better changes: 

image

Another option would be going to `github.dev` to review this. 
to go to `github.dev` you need to do one of the next:
- click `.` dot in the keyboard
- change the `.com` in the URL to `.dev`

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

fixes #1448

## Mobile & Desktop Screenshots/Recordings

The checkbox elements and labels are connected to their related inputs:
image

To Test you need to do the next: 
- Go to: https://insights.opensauced.pizza/user/settings 
- Click on the check boxes labels one and see if their related check box is triggered(display email, collaboration, timezone)

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by OgDev-01 — APPROVED** [review]
LGTM

**Review by ppaul — COMMENTED** [review]
there will be 2 calls of getId inside the JSX. I would rather suggest determining the id only once in a const, then return the JSX.

File: components/atoms/Checkbox/checkbox.tsx [hunk 1]
```diff
@@ -9,31 +9,34 @@ interface CheckboxProps extends React.ComponentPropsWithoutRef<typeof CheckboxPr
...
-          htmlFor={id}
-          className="ml-3 text-sm font-medium leading-none cursor-pointer text-light-slate-12 peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
+  ({ className, label, id, ...props }, ref) => {
+    const getId = () => (id ? id : label?.replaceAll(" ", "_").toLowerCase());
+    return (
+      <div className="flex items-center">
+        <CheckboxPrimitive.Root
+          ref={ref}
+          className={clsx(
+            "peer h-4 w-4 shrink-0 rounded-[4px] cursor-pointer bg-white border border-light-slate-8 hover:border-orange-500 focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:border-orange-500 data-[state=checked]:bg-orange-500",
```
Comment (NsdHSO) [inline components/atoms/Checkbox/checkbox.tsx:19]:
```suggestion
            "peer h-4 w-4 shrink-0 rounded cursor-pointer bg-white border border-light-slate-8 hover:border-orange-500 focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:border-orange-500 data-[state=checked]:bg-orange-500",
```

Comment (a0m0rajab) [reply components/atoms/Checkbox/checkbox.tsx]: Thank you for your comment!

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**